### PR TITLE
GEOSEARCH ANY argument

### PR DESCRIPTION
**Pull request by oranagra**

### CODE REVIEW

File: commands/georadius.md [hunk 1]
```diff
@@ -22,7 +22,10 @@ The command default is to return unsorted items. Two different sorting methods c
 * `ASC`: Sort returned items from the nearest to the farthest, relative to the center.
 * `DESC`: Sort returned items from the farthest to the nearest, relative to the center.
 
-By default all the matching items are returned. It is possible to limit the results to the first N matching items by using the **COUNT `<count>`** option. However note that internally the command needs to perform an effort proportional to the number of items matching the specified area, so to query very large areas with a very small `COUNT` option may be slow even if just a few results are returned. On the other hand `COUNT` can be a very effective way to reduce bandwidth usage if normally just the first results are used.
+By default all the matching items are returned. It is possible to limit the results to the first N matching items by using the **COUNT `<count>`** option.
+When `<count>` is positive, the command will internally perform an effort proportional to the number of items matching the specified area and sort sort them,
```
Comment (itamarhaber) [inline commands/georadius.md:26]:
```suggestion
When `<count>` is positive, the command will perform an effort that is proportional to the number of items matching the specified area and sort them,
```

File: commands/georadius.md [hunk 2]
```diff
@@ -22,7 +22,10 @@ The command default is to return unsorted items. Two different sorting methods c
...
+so to query very large areas with a very small `COUNT` option may be slow even if just a few results are returned.
+When `<count>` is negative the command will return ASAP as soon as enough matches are found, so the results that return are not necessarily the best matches.
```
Comment (itamarhaber) [inline commands/georadius.md:28]:
```suggestion
When `<count>` is negative the command will return as soon as enough matches are found. That means the results may not be the ones closest to the specified point, but on the other hand, the effort invested by the server is significantly lower.
```

File: commands/georadius.md [hunk 3]
```diff
@@ -54,6 +57,10 @@ Breaking the compatibility with the past was considered but rejected, at least f
 
 Both commands were introduced in Redis 3.2.10 and Redis 4.0.0 respectively.
 
+@history
+
+* `>= 6.2`: COUNT can take negative number to return some matches ASAP.
```
Comment (itamarhaber) [inline commands/georadius.md:62]:
```suggestion
* `>= 6.2`: COUNT accepts a negative value and returns as soon as possible.
```

File: commands/geosearch.md [hunk 4]
```diff
@@ -21,7 +21,10 @@ By default, the command returns unsorted items. Two different sorting methods ca
 * `ASC`: Sort returned items from the nearest to the farthest, relative to the center.
 * `DESC`: Sort returned items from the farthest to the nearest, relative to the center.
 
-By default, all the matching items are returned. It is possible to limit the results to the first N matching items by using the **COUNT `<count>`** option. However note that internally the command needs to perform an effort proportional to the number of items matching the specified area, so to query very large areas with a very small `COUNT` option may be slow even if just a few results are returned. On the other hand `COUNT` can be a very effective way to reduce bandwidth usage if normally just the first results are used.
+By default all the matching items are returned. It is possible to limit the results to the first N matching items by using the **COUNT `<count>`** option.
+When `<count>` is positive, the command will internally perform an effort proportional to the number of items matching the specified area and sort sort them,
```
Comment (itamarhaber) [inline commands/geosearch.md:25]:
```suggestion
When `<count>` is positive, the command will perform an effort that is proportional to the number of items matching the specified area and sort them,
```

File: commands/geosearch.md [hunk 5]
```diff
@@ -21,7 +21,10 @@ By default, the command returns unsorted items. Two different sorting methods ca
...
+so to query very large areas with a very small `COUNT` option may be slow even if just a few results are returned.
+When `<count>` is negative the command will return ASAP as soon as enough matches are found, so the results that return are not necessarily the best matches.
```
Comment (itamarhaber) [inline commands/geosearch.md:27]:
```suggestion
When `<count>` is negative the command will return as soon as enough matches are found. That means the results may not be the ones closest to the specified point, but on the other hand, the effort invested by the server is significantly lower.
```

File: commands/geosearch.md [hunk 6]
```diff
@@ -36,6 +39,10 @@ When additional information is returned as an array of arrays for each item, the
 2. The geohash integer.
 3. The coordinates as a two items x,y array (longitude,latitude).
 
+@history
+
+* `>= 6.2`: COUNT can take negative number to return some matches ASAP.
```
Comment (itamarhaber) [inline commands/geosearch.md:44]:
```suggestion
* `>= 6.2`: COUNT accepts a negative value and returns as soon as possible.
```

Comment (oranagra) [reply commands/geosearch.md:44]: actually GEOSARCH is a new command, we need to drop this part

File: commands/geosearch.md [hunk 7]
```diff
@@ -36,6 +39,10 @@ When additional information is returned as an array of arrays for each item, the
...
+
```
Comment (oranagra) [inline commands/geosearch.md:45]:
```suggestion
```

**Review by itamarhaber — APPROVED** [review]

**Comment by oranagra** [comment]
@itamarhaber please review my last commit (and also the recent changes in the redis code itself (the error message).
note that the command actually takes the ANY argument at any position, but in the docs in embedded it into the COUNT.
let me know if you think that's ok

**Comment by oranagra** [comment]
can't figure out why it fails to parse the JSON file

**Comment by itamarhaber** [comment]
Thanks @oranagra - reviewing. 

> can't figure out why it fails to parse the JSON file

Yeah, you have a "dangling" comma:

https://github.com/redis/redis-doc/pull/1480/files#diff-1b67c587a1e3cd259a83610a34d5b7f57e9702605608178aac699085891c07e9R1407

File: commands.json [hunk 8]
```diff
@@ -1398,9 +1398,23 @@
...
       {
-        "command": "COUNT",
+        "type": "block",
         "name": "count",
-        "type": "integer",
+        "block": [
+          {
+            "name": "count",
+            "command": "COUNT",
+            "type": "integer",
```
Comment (itamarhaber) [inline commands.json:1407]:
```suggestion
            "type": "integer"
```

File: commands.json [hunk 9]
```diff
@@ -1479,9 +1493,23 @@
...
```
Comment (itamarhaber) [inline commands.json:1502]:
```suggestion
            "type": "integer"
```

File: commands.json [hunk 10]
```diff
@@ -1593,9 +1621,23 @@
...
       {
+        "type": "block",
         "name": "count",
-        "command": "COUNT",
-        "type": "integer",
+        "block": [
+          {
+            "name": "count",
+            "command": "COUNT",
+            "type": "integer",
```
Comment (itamarhaber) [inline commands.json:1630]:
```suggestion
            "type": "integer"
```

File: commands.json [hunk 11]
```diff
@@ -1714,9 +1756,23 @@
...
```
Comment (itamarhaber) [inline commands.json:1765]:
```suggestion
            "type": "integer"
```

**Pull request merged**